addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -259,6 +259,8 @@ def _compute_number_of_days(self):
             if allocation_unit != 'hour':
                 allocation.number_of_days = allocation.number_of_days_display
             else:
+                if not allocation.employee_id._get_hours_per_day(allocation.date_from):
```

Comment:
Store the call `allocation.employee_id._get_hours_per_day(allocation.date_from)` in a variable so you don't repeat it 2 lines below